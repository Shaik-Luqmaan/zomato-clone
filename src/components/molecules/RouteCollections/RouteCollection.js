import React from 'react'
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import Typography from "../../atoms/Typography/Typography";

const useStyles = makeStyles({
    root: {
      border: "1px solid rgb(0,0,0,.2)",
      borderRadius: "15px",
      maxWidth: "280px",
      cursor: "pointer",
      transition: "transform .5s",
      "&:hover": {
        transform: "scale(1.08)",
      },
    },
    img: {
      objectFit: "cover",
      width: "100%",
      height: "180px",
      borderRadius: "15px 15px 0 0",
    },
    text: {
      padding: "8px 8px 15px 15px",
      color: "rgb(0,0,0,.8)",
    },
  });

const RouteCollection = ({ img, title, ...props }) => {
    const classes = useStyles();
    return (
    <Grid
        container
        direction="column"
        alignItems="center"
        className={classes.root}
        {...props}
     >
      <Grid item>
        <img alt="Route Card" 
            className={classes.img} 
            src={img} />
      </Grid>

      <Grid item className={classes.text}>
        <Typography variant="h6" data-testid="sample">
          {title}
        </Typography>
      </Grid>

    </Grid>
    )
}

export default RouteCollection
