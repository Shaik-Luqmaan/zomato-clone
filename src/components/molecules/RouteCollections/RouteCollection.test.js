import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom"
import renderer from 'react-test-renderer';
import RouteCollection from './RouteCollection';

afterEach(cleanup);

it("renders component without crashing",()=>{
    const div = document.createElement('div');
    ReactDOM.render(<RouteCollection />, div)
})

it("matches snapshot",()=>{
    const tree = renderer.create(<RouteCollection/>).toJSON();
    expect(tree).toMatchSnapshot();
})

it("render component correctly",()=>{
    const args ={
        title:"Order Food Online",
        img: "https://b.zmtcdn.com/webFrontend/64dffaa58ffa55a377cdf42b6a690e721585809275.png"
    }
  
    const{getByTestId} = render(<RouteCollection {...args}/>);
    expect(getByTestId("sample")).toHaveTextContent(args.title);
})

it("matches snapshot 2",()=>{
    const args ={
        title:"Order Food Online",
        img: "https://b.zmtcdn.com/webFrontend/64dffaa58ffa55a377cdf42b6a690e721585809275.png"
    }
    const tree = renderer.create(<RouteCollection {...args}/>).toJSON();
    expect(tree).toMatchSnapshot();
})
