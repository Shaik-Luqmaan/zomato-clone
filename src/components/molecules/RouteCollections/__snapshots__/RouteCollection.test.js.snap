// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <img
      alt="Route Card"
      className="makeStyles-img-2"
    />
  </div>
  <div
    className="MuiGrid-root makeStyles-text-3 MuiGrid-item"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="sample"
    />
  </div>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <img
      alt="Route Card"
      className="makeStyles-img-2"
      src="https://b.zmtcdn.com/webFrontend/64dffaa58ffa55a377cdf42b6a690e721585809275.png"
    />
  </div>
  <div
    className="MuiGrid-root makeStyles-text-3 MuiGrid-item"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6"
      data-testid="sample"
    >
      Order Food Online
    </h6>
  </div>
</div>
`;
